data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessorConfig.java

```diff
@@ -5,50 +5,90 @@
 package org.opensearch.dataprepper.plugins.lambda.processor;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
 import jakarta.validation.Valid;
 import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 import jakarta.validation.constraints.Size;
+import org.opensearch.dataprepper.model.configuration.PluginModel;
 import org.opensearch.dataprepper.plugins.lambda.common.config.AwsAuthenticationOptions;
 import org.opensearch.dataprepper.plugins.lambda.common.config.BatchOptions;
-import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.BATCH_EVENT;
+import org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig;
+import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.DEFAULT_CONNECTION_RETRIES;
+import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.DEFAULT_SDK_TIMEOUT;
 import static org.opensearch.dataprepper.plugins.lambda.common.config.LambdaCommonConfig.REQUEST_RESPONSE;
 
-public class LambdaProcessorConfig {
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
 
-    private static final int DEFAULT_CONNECTION_RETRIES = 3;
+public class LambdaProcessorConfig {
+    //Ensures 1:1 mapping of events input to lambda and response from lambda
+    public static final String STRICT = "strict";
```

Comment:
Please use enums for these. This is important for correct support for validations and schemas.

Here is an example:

https://github.com/opensearch-project/data-prepper/blob/27337ea36ee7a8a4b3496b32a46c541abd9aa609/data-prepper-plugins/decompress-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/decompress/DecompressionType.java